app/scripts/background.js

```diff
@@ -426,7 +430,10 @@ function setupController(initState, initLangCode) {
   pump(
     storeAsStream(controller.store),
     debounce(1000),
-    createStreamSink((state) => localStore.set(state)),
+    createStreamSink(async (state) => {
+      await localStore.set(state);
+      statePersistenceEvents.emit('state-persisted', state);
```

Comment:
We thought this was harmless to add without the need of an override, as we are just emitting an event for anyone that might be interested about a state change.

Happy to discuss otherwise.